assets/dev/js/editor/controls/media.js

```diff
@@ -222,27 +254,79 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		}
 
 		if ( attachment.url ) {
-			this.setValue( {
+			const updatedState = {
 				url: attachment.url,
 				id: attachment.id,
 				alt: attachment.alt,
 				source: attachment.source,
-			} );
+			};
+			this.setValue( updatedState );
 
 			if ( this.model.get( 'responsive' ) ) {
 				// Render is already calls `applySavedValue`, therefore there's no need for it in this case.
 				this.renderWithChildren();
 			} else {
 				this.applySavedValue();
 			}
-		}
 
+			this.updatePrevState( updatedState );
+			this.toggleSizeControl();
+		}
 		this.trigger( 'after:select' );
 	},
 
 	onBeforeDestroy() {
 		this.$el.remove();
 	},
+
+	async onMediaImageSize() {
+		this.ui.mediaImageSize.removeClass( 'e-select-placeholder' );
+		const selectedImage = {
+			id: this.getEditSettings( 'mediaId' ),
+			alt: this.getEditSettings( 'mediaAlt' ),
+		};
+
+		const selectedSize = this.ui.mediaImageSize.val();
+		let imageURL = await elementor.imagesManager.getImageUrl( {
+			id: selectedImage.id,
+			size: selectedSize,
+		} );
+
+		const stateOptions = {
+			url: null,
+			id: selectedImage.id,
+			alt: selectedImage.alt,
+			dimensions: selectedSize,
+			source: 'library',
+		};
+
+		if ( imageURL ) {
+			stateOptions.url = imageURL;
+			this.setValue( stateOptions );
+		} else {
+			elementor.channels.editor.once( 'imagesManager:detailsReceived', ( data ) => {
+				imageURL = data[ selectedImage.id ][ selectedSize ];
+				if ( imageURL ) {
+					stateOptions.url = imageURL;
+					this.setValue( stateOptions );
+				}
+			} );
+		}
+	},
+
+	toggleSizeControl() {
+		const sizesSupport = this.model.get( 'sizes_supported' );
+		const hasImage = this.getEditSettings( 'image_url' );
+		const opacity = this.getEditSettings( 'image_opacity' );
```

Comment:
`hasImage` & `opacity` don't really say a lot. maybe we should try finding better names (same as the `setEditSettings()` we need to change above).

And regarding the "sizes support", maybe let's ask @matipojo if there's any naming convention for that

Reply:
Vars are renamed from previews review.